### PR TITLE
fix: Regex to detect function args

### LINKED ISSUE #8169: [Bug] GUI builder for Navigate-To query params updates with random input
## Description

Trying to use the GUI option to write query params results in unexpected/random output. More info in the attached video

### Steps to reproduce the behavior:

This is shown in the attached video

### Important Details

- Version: Cloud
- OS: macOS Big Sur
- Browser: Edge
- Environment: Release

https://user-images.githubusercontent.com/17744578/135872049-0bf20b08-d2b8-4ce3-9c24-a8b00117a89f.mov

### PR DESCRIPTION
**Pull request by aswathkk**
## Description

Regex was not working with whitespace characters since we are using `.*`
Replaced it with `[\s\S]*` to include whitespace characters.
Arrow function was not matching when space is there between arrow (`()  =>  {}`) that is also fixed.

Fixes #8169

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

Tested this manually by providing whitespace characters into the argument fields of action selectors.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:green_circle: Total coverage has increased</summary>

    // Code coverage diff between base branch:release and head branch: fix/action-selector-issue 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :green_circle: | total | 55.15 **(0.02)** | 37.06 **(0.01)** | 34.04 **(0.02)** | 55.67 **(0.02)**
 :green_circle: | app/client/src/components/editorComponents/ActionCreator/Fields.tsx | 35.52 **(3.78)** | 25.42 **(3.39)** | 10.39 **(2.6)** | 34.86 **(3.67)**
 :red_circle: | app/client/src/utils/autocomplete/TernServer.ts | 52.71 **(-0.23)** | 40.83 **(-0.84)** | 36.21 **(0)** | 56.74 **(-0.25)**</details>

### CODE REVIEW

File: app/client/src/components/editorComponents/ActionCreator/Fields.tsx [hunk 1]
```diff
@@ -73,7 +73,7 @@ const NAVIGATION_TARGET_FIELD_OPTIONS = [
   },
 ];
 
-export const FUNC_ARGS_REGEX = /((["][^"]*["])|([\[].*[\]])|([\{].*[\}])|(['][^']*['])|([\(].*[\)[=][>][{].*[}])|([^'",][^,"+]*[^'",]*))*/gi;
+export const FUNC_ARGS_REGEX = /((["][^"]*["])|([\[][\s\S]*[\]])|([\{][\s\S]*[\}])|(['][^']*['])|([\(][\s\S]*[\)[=][>][{][\s\S]*[}])|([^'",][^,"+]*[^'",]*))*/gi;
```
Comment (hetunandu) [inline app/client/src/components/editorComponents/ActionCreator/Fields.tsx:76]: Needs a test

**Review by hetunandu — CHANGES_REQUESTED** [review]

**Review by hetunandu — APPROVED** [review]

**Comment by vivekverma2312** [comment]
@aswathkk Deploy preview is tested and working fine.

**Review by hetunandu — APPROVED** [review]

**Pull request merged**